sashimi-webapp/src/components/file-manager/File.vue

```diff
@@ -45,6 +55,18 @@ export default {
 
       this.file.rename(newFileName);
     },
+    onClick(event) {
+      if (this.onClickDetails.isClickInProgress && event.x === this.onClickDetails.x && event.y === this.onClickDetails.y) {
```

Comment:
`event.x === this.onClickDetails.x && event.y === this.onClickDetails.y` require the subsequent click to be in the exact same location.

This is very difficult for touch device as the finger is not likely to hit the exact coordinate.

You may either implement a threshold area or remove this check.

Threshold area e.g.
```javascript
dblClickCheck(event) {
  const threshold = 20;
  return (this.onClickDetails.isClickInProgress &&
          ((Math.abs(event.x - this.onClickDetails.x) < threshold) &&
            (Math.abs(event.y - this.onClickDetails.y) < threshold)
          ));
},
onClick(event) {
  if (this.dblClickCheck(event)) {
    this.openFile();
    this.clearIsClick();
    clearTimeout(this.clearIsClick);
  } else {
    this.onClickDetails.x = event.x;
    this.onClickDetails.y = event.y;
    this.onClickDetails.isClickInProgress = true;
    setTimeout(this.clearIsClick, 1000);
  }
},
```

I chose a threshold of 20 based on a quick experiment I conducted with my 5.5 inches screen Android phone.
Result showing data for 138 touches.

20px is equivalent to:

|Device|Screen Size (inches) |Actual (mm)|
---|---|---
Oneplus 3| 5.5 | 4.50
iPhone 5 | 5 | 3.47
Nexus 6P | 5.7 | 3.46
Nexus 5X | 5.2 | 3.16

![](http://i.imgur.com/MWqDo6v.png) 

<img src="http://i.imgur.com/23M57tP.png" width="360">